inference-engine/tests/unit/gna/ngraph/transformations/gna_remove_single_input_concat.cpp

```diff
@@ -0,0 +1,140 @@
+                                                                                              outputs.end()),
+                                                                         0);
+        outputs = {concat_operation};
+    }
+
+    for (auto output : outputs) {
+        Operations new_outputs;
+        for (int i = 0; i < n_outputs; ++i) {
+            auto add_bias = ngraph::opset7::Constant::create(ngraph::element::i64, {1, 1, 1}, {3});
+            auto add_operation = std::make_shared<ngraph::opset7::Add>(output, add_bias);
+        }
+        outputs.swap(new_outputs);
+    }
+
+    Graph graph;
+    graph.inputs.swap(inputs);
+    graph.outputs.insert(graph.outputs.end(),
+                        std::make_move_iterator(graph.outputs.begin()),
+                        std::make_move_iterator(graph.outputs.end()));
+
+    return graph;
+}
+
+// -------------------------------------------------------------------------------------------------------
+
+class RemoveSingleInputConcatFixture: public CommonTestUtils::TestsCommon,
+                               public ::testing::WithParamInterface<std::tuple<Graph /* tranformed */,
+                                                                               Graph /* reference */>> {
+public:
+    void SetUp() override;
+public:
+    std::shared_ptr<ngraph::Function> function, reference_function;
+};
+
+void RemoveSingleInputConcatFixture::SetUp() {
+    // TODO: use auto & [transformed_graph, reference_graph] = this->GetParam() when C++17
+    Graph transformed_graph;
+    Graph reference_graph;
+    std::tie(transformed_graph, reference_graph) = this->GetParam();
+
+    function = transformed_graph.createFunction();
+    reference_function = reference_graph.createFunction();
+}
+
+ngraph::pass::Manager createPassManager() {
+    ngraph::pass::Manager manager;
+    manager.register_pass<ngraph::pass::InitNodeInfo>();
+    manager.register_pass<GNAPluginNS::RemoveSingleInputConcat>();
+    return manager;
+}
+
+void execute_test(std::shared_ptr<ngraph::Function> function,
+                  std::shared_ptr<ngraph::Function> reference_function) {
+    ngraph::pass::Manager pass_manager = createPassManager();
+    pass_manager.run_passes(function);
+    const FunctionsComparator func_comparator = FunctionsComparator::with_default().enable(FunctionsComparator::ATTRIBUTES);
+    const FunctionsComparator::Result result = func_comparator(function, reference_function);
+    ASSERT_TRUE(result.valid);
+}
+
+TEST_P(RemoveSingleInputConcatFixture, CompareFunctions) {
+    execute_test(function, reference_function);
+}
+
+INSTANTIATE_TEST_SUITE_P(RemoveSingleInputConcatTestSuite, RemoveSingleInputConcatFixture,
+                         ::testing::Values(std::make_tuple(createGraph(1 /* n_inputs */, true /* has_concat */, 1 /* n_outputs */),
+                                                           createGraph(1 /* n_inputs */, true /* has_concat */, 1 /* n_outputs */)),
```

Comment:
Why do we have a `true` here? Concat with 1 input and 1 output has no sense and should be removed.